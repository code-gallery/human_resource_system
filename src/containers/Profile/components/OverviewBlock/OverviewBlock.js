import React from 'react'
import PropTypes from 'prop-types'
import Overview from './components/Overview'
import Activity from './components/Activity'
import EditProfile from './components/EditProfile'

const OverviewBlock = (props) => {
  const {
    editMode,
    user,
    onChangeProfileField,
    activities,
    ...rest
  } = props

  const invalidFields = []
  if (user.first_name === '') {
    invalidFields.push('first_name')
  }
  if (user.last_name === '') {
    invalidFields.push('last_name')
  }

  return (
    <div className="col-xs-12 col-md-4">
      <div className="hidden-sm hidden-xs">
        {editMode &&
          <EditProfile
            onChangeProfileField={onChangeProfileField}
            user={user}
            invalidFields={invalidFields}
          />
        }
      </div>
      <Overview
        {...rest}
        editMode={editMode}
      />
      <Activity
        activities={activities}
        editMode={editMode}
      />
    </div>
  )
}

OverviewBlock.propTypes = {
  editMode: PropTypes.bool,
  user: PropTypes.object,
  onChangeProfileField: PropTypes.func,
  activities: PropTypes.array
}

export default OverviewBlock
