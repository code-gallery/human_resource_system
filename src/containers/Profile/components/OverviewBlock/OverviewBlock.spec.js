import React from 'react'
import renderer from 'react-test-renderer'
import OverviewBlock from './OverviewBlock'

describe('containers/Profile/components/OverviewBlock', () => {
  const commonProps = {
    user: {
      first_name: 'John',
      last_name: 'Doe',
      public_email: 'johndoe@appliedblockchain.com',
      location: 'London, UK'
    },
    jobs: [
      { verified: true },
      { verified: false },
      { verified: false }
    ],
    educations: [
      { verified: true },
      { verified: true },
      { verified: false }
    ],
    allAwards: {
      achievement: [],
      award: [
        { verified: true },
        { verified: false }
      ],
      certificate: [],
      cpd: [
        { verified: true },
        { verified: true },
        { verified: true }
      ],
      language: [
        { verified: true },
        { verified: false }
      ],
      project: [
        { verified: true },
        { verified: true },
        { verified: true }
      ],
      skill: []
    },
    activities: [],
    onChangeProfileField: jest.fn()
  }

  describe('when editMode is false', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        editMode: false
      }
      const tree = renderer.create(
        <OverviewBlock {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when editMode is true', () => {
    describe('when name is valid', () => {
      it('renders correctly', () => {
        const props = {
          ...commonProps,
          editMode: true
        }
        const tree = renderer.create(
          <OverviewBlock {...props} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    describe('when name is invalid', () => {
      it('renders correctly', () => {
        const props = {
          ...commonProps,
          user: {
            first_name: '',
            last_name: ''
          },
          editMode: true
        }
        const tree = renderer.create(
          <OverviewBlock {...props} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })
})
