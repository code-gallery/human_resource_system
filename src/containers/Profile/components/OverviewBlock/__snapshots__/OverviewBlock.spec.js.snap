// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/OverviewBlock when editMode is false renders correctly 1`] = `
<div
  className="col-xs-12 col-md-4"
>
  <div
    className="hidden-sm hidden-xs"
  />
  <div
    className="profile-overview "
  >
    <p
      className="profile-block-title"
    >
      Overview
    </p>
    <ul
      className="overview-info"
    >
      <li>
        <div
          className="progress-item progress-67"
        >
          <div
            className="radial-inner-bg"
          >
            2/3
          </div>
        </div>
        <span>
          Education
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-33"
        >
          <div
            className="radial-inner-bg"
          >
            1/3
          </div>
        </div>
        <span>
          Work Experience
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-50"
        >
          <div
            className="radial-inner-bg"
          >
            1/2
          </div>
        </div>
        <span>
          Awards
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-100"
        >
          <div
            className="radial-inner-bg"
          >
            3/3
          </div>
        </div>
        <span>
          CPD
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-100"
        >
          <div
            className="radial-inner-bg"
          >
            3/3
          </div>
        </div>
        <span>
          Project
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-0"
        >
          <div
            className="radial-inner-bg"
          >
            0/0
          </div>
        </div>
        <span>
          Certificate
        </span>
      </li>
    </ul>
    <div
      className="profile-verified-block"
    >
      <p>
        <span>
          71% verified
        </span>
        <i
          className="tooltip-position-top tooltip-movable tooltip-dark"
          data-tooltip="10/14 claims verified"
        />
      </p>
      <div
        className="profile-verified-holder"
      >
        <div
          className="profile-verified-bar"
          style={
            Object {
              "width": "71%",
            }
          }
        >
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="Activity "
  >
    <p
      className="Activity-title"
    >
      Activity
    </p>
    <ul>
      <p>
        No activities yet
      </p>
    </ul>
  </div>
</div>
`;

exports[`containers/Profile/components/OverviewBlock when editMode is true when name is invalid renders correctly 1`] = `
<div
  className="col-xs-12 col-md-4"
>
  <div
    className="hidden-sm hidden-xs"
  >
    <div
      className="profile-overview-edit"
    >
      <p
        className="profile-block-title"
      >
        Details
      </p>
      <div
        className="form-group"
      >
        <label
          className="required-error"
        >
          First Name
        </label>
        <input
          defaultValue=""
          name="first_name"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="required-error"
        >
          Last Name
        </label>
        <input
          defaultValue=""
          name="last_name"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          About
        </label>
        <textarea
          name="tagline"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <div>
          <input
            placeholder="Enter your address"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label>
          Twitter
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Public Email
        </label>
        <input
          defaultValue={undefined}
          name="public_email"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Public Telephone
        </label>
        <input
          defaultValue={undefined}
          name="mobile"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Date of Birth
        </label>
        <DateTimeMock
          dateFormat="DD/MM/YYYY"
          name="dob"
          onChange={[Function]}
          timeFormat={false}
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="profile-overview edit-mode"
  >
    <p
      className="profile-block-title"
    >
      Overview
    </p>
    <ul
      className="overview-info"
    >
      <li>
        <div
          className="progress-item progress-67"
        >
          <div
            className="radial-inner-bg"
          >
            2/3
          </div>
        </div>
        <span>
          Education
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-33"
        >
          <div
            className="radial-inner-bg"
          >
            1/3
          </div>
        </div>
        <span>
          Work Experience
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-50"
        >
          <div
            className="radial-inner-bg"
          >
            1/2
          </div>
        </div>
        <span>
          Awards
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-100"
        >
          <div
            className="radial-inner-bg"
          >
            3/3
          </div>
        </div>
        <span>
          CPD
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-100"
        >
          <div
            className="radial-inner-bg"
          >
            3/3
          </div>
        </div>
        <span>
          Project
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-0"
        >
          <div
            className="radial-inner-bg"
          >
            0/0
          </div>
        </div>
        <span>
          Certificate
        </span>
      </li>
    </ul>
    <div
      className="profile-verified-block"
    >
      <p>
        <span>
          71% verified
        </span>
        <i
          className="tooltip-position-top tooltip-movable tooltip-dark"
          data-tooltip="10/14 claims verified"
        />
      </p>
      <div
        className="profile-verified-holder"
      >
        <div
          className="profile-verified-bar"
          style={
            Object {
              "width": "71%",
            }
          }
        >
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="Activity edit-mode"
  >
    <p
      className="Activity-title"
    >
      Activity
    </p>
    <ul>
      <p>
        No activities yet
      </p>
    </ul>
  </div>
</div>
`;

exports[`containers/Profile/components/OverviewBlock when editMode is true when name is valid renders correctly 1`] = `
<div
  className="col-xs-12 col-md-4"
>
  <div
    className="hidden-sm hidden-xs"
  >
    <div
      className="profile-overview-edit"
    >
      <p
        className="profile-block-title"
      >
        Details
      </p>
      <div
        className="form-group"
      >
        <label
          className=""
        >
          First Name
        </label>
        <input
          defaultValue="John"
          name="first_name"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Last Name
        </label>
        <input
          defaultValue="Doe"
          name="last_name"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          About
        </label>
        <textarea
          name="tagline"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <div>
          <input
            placeholder="Enter your address"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label>
          Twitter
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Public Email
        </label>
        <input
          defaultValue="johndoe@appliedblockchain.com"
          name="public_email"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Public Telephone
        </label>
        <input
          defaultValue={undefined}
          name="mobile"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Date of Birth
        </label>
        <DateTimeMock
          dateFormat="DD/MM/YYYY"
          name="dob"
          onChange={[Function]}
          timeFormat={false}
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="profile-overview edit-mode"
  >
    <p
      className="profile-block-title"
    >
      Overview
    </p>
    <ul
      className="overview-info"
    >
      <li>
        <div
          className="progress-item progress-67"
        >
          <div
            className="radial-inner-bg"
          >
            2/3
          </div>
        </div>
        <span>
          Education
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-33"
        >
          <div
            className="radial-inner-bg"
          >
            1/3
          </div>
        </div>
        <span>
          Work Experience
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-50"
        >
          <div
            className="radial-inner-bg"
          >
            1/2
          </div>
        </div>
        <span>
          Awards
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-100"
        >
          <div
            className="radial-inner-bg"
          >
            3/3
          </div>
        </div>
        <span>
          CPD
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-100"
        >
          <div
            className="radial-inner-bg"
          >
            3/3
          </div>
        </div>
        <span>
          Project
        </span>
      </li>
      <li>
        <div
          className="progress-item progress-0"
        >
          <div
            className="radial-inner-bg"
          >
            0/0
          </div>
        </div>
        <span>
          Certificate
        </span>
      </li>
    </ul>
    <div
      className="profile-verified-block"
    >
      <p>
        <span>
          71% verified
        </span>
        <i
          className="tooltip-position-top tooltip-movable tooltip-dark"
          data-tooltip="10/14 claims verified"
        />
      </p>
      <div
        className="profile-verified-holder"
      >
        <div
          className="profile-verified-bar"
          style={
            Object {
              "width": "71%",
            }
          }
        >
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="Activity edit-mode"
  >
    <p
      className="Activity-title"
    >
      Activity
    </p>
    <ul>
      <p>
        No activities yet
      </p>
    </ul>
  </div>
</div>
`;
